bin/population/pycbc_multiifo_pastro

```diff
@@ -0,0 +1,396 @@
+    logging.warn('Adding events from', args.prior_event_info)
+    prior_info = np.genfromtxt(args.prior_event_info, names=True)
+    allfgbg = np.append(allfgbg, prior_info['ln_bayes_factor'])
+else:  # no prior events!
+    prior_info = {'stat': np.array([]),
+                  'ifar': np.array([]),
+                  'gps': np.array([]),
+                  'bayes_factor': np.array([])}
+
+# do the calculation
+foreground_count_dist = \
+      fgmcl.count_posterior(allfgbg, laguerre_n=args.laguerre_degree,
+                            Lambda0=bg.norm, name='foreground count posterior',
+                            prior=args.power_law_prior)
+
+# Roulet et al. argue that information on rate scales with sum (p_astro^2)
+logging.info('sum of p_astro^2',
+             ((1 - foreground_count_dist.p_bg(allfgbg)) ** 2.).sum())
+
+fgmc_plots.odds_summary(
+    args,
+    np.append(fg.stat, prior_info['stat']),
+    np.append(fg.ifar, prior_info['ifar']),
+    foreground_count_dist.p_bg(allfgbg),
+    args.ntop,
+    times=np.append(fg.gpst, prior_info['gps']),
+    name='foreground events',
+    plot_extensions=['.png'],
+)
+
+args.plot_limits = None
+cmed, cminus, cplus = fgmc_plots.dist_summary(
+    args,
+    foreground_count_dist,
+    plot_extensions=['.png'],
+    middle='median',
+    credible_intervals=[0.90],
+)
+
+# if VT and error are provided, convert signal PDF to rate PDF
+if args.vt_mean:
+    if args.vt_err is None:
+        logging.warn('NB : no VT error provided, will use 0!')
+        args.vt_err = 0.
+    logging.info('Rate in units of ' + args.rate_units + ' ...')
+    print('Rate without VT uncertainty :')
+    print('%.1f_{-%.1f}^{+%.1f}' % (
+        cmed / args.vt_mean, abs(cminus) / args.vt_mean, cplus / args.vt_mean))
+    import rate_functions
+    fracerr = args.vt_err / args.vt_mean
+    # attr that can arbitrarily rescale counts in rate calculation : don't do this
+    foreground_count_dist.scale = 1.
+    # we want a Jeffreys type prior: this is already supplied by the default value
+    # of the --power-law-prior option (see above) so leave a uniform prior on rate
+    rate_distribution = \
+        rate_functions.rate_posterior(foreground_count_dist, args.vt_mean,
+                                      fracerr, unit=args.rate_units, texunit='')
+    median, minus, plus = fgmc_plots.dist_summary(args, rate_distribution,
+          plot_extensions=None, credible_intervals=[0.9])
+    print('Rate with VT uncertainty via rate_posterior :')
+    print('%.1f_{-%.1f}^{+%.1f}' % (median, abs(minus), plus))
+elif args.vt_err:
+    raise RuntimeError("Can't supply VT error without a VT mean!")
+
+
+# if injections are to be evaluated 'like zerolag', read them in
+if args.rank_injections:
+    assert args.p_astro_inj is not None
+    inj = utils.ForegroundEvents(args, args.coinc_times,
+                                 bin_lo=args.min_mchirp, bin_hi=args.max_mchirp)
+
+    # assume that we use the same bank all through
```

Comment:
Is this a safe assumption? If this is a safe assumption, should this be implemented for zerolag?

Reply:
maybe ...